cmd/anonymizer/app/query/query_test.go

```diff
@@ -48,6 +48,59 @@ var (
 	}
 )
 
+const (
+	maxSpanCountInChunk = 10
+	msgTraceNotFound    = "trace not found"
+)
+
+var errUninitializedTraceID = status.Error(codes.InvalidArgument, "uninitialized TraceID is not allowed")
+
+// testGRPCHandler is a minimal implementation of api_v2.QueryServiceServer
+// for testing purposes. It only implements GetTrace, using the embedded
+// UnimplementedQueryServiceServer for other methods.
+type testGRPCHandler struct {
+	api_v2.UnimplementedQueryServiceServer
+	spanReader spanstore.Reader
+}
+
+// GetTrace implements the gRPC GetTrace method by reading from the span reader.
+// This is a minimal implementation copied from cmd/query/app/grpc_handler.go
+// for the purpose of the anonymizer test.
+func (g *testGRPCHandler) GetTrace(r *api_v2.GetTraceRequest, stream api_v2.QueryService_GetTraceServer) error {
+	if r.TraceID == (model.TraceID{}) {
+		return errUninitializedTraceID
+	}
+	query := spanstore.GetTraceParameters{
+		TraceID:   r.TraceID,
+		StartTime: r.StartTime,
+		EndTime:   r.EndTime,
+	}
+	trace, err := g.spanReader.GetTrace(stream.Context(), query)
```

Comment:
Why use "real" spanReader when the other test mocks it as well? We can just make this grpc server return expected result directly.